dex/farm/src/lib.rs

```diff
@@ -159,7 +159,7 @@ pub trait Farm:
             &farm_contribution,
             &farm_token_id,
             &attributes,
-            &self.manage_vec_remove_index(&payments, 0),
+            payments_iter,
```

Comment:
Just a double check. The .next() method also increases your iter position? otherwise you might need to skip it manually

Reply:
yes